### PR TITLE
Type checker integration tests

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This adds integration tests for the type checker added in https://github.com/pulumi/pulumi-terraform-bridge/pull/2000
Not all the cases can be tested still, as some are caught by the language runtime. Will look into that more to expand the feature flag for the YAML runtime.

Depends on a debug feature flag for the YAML runtime from https://github.com/pulumi/pulumi-yaml/pull/634

fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/2418

### CODE REVIEW

**Review by t0yv0 — APPROVED** [review]
Very nice. Adding coverage is much needed.

File: pkg/tests/schema_pulumi_test.go [hunk 1]
```diff
@@ -4159,3 +4159,100 @@ func TestMakeTerraformResultNilVsEmptyMap(t *testing.T) {
...
+resources:
+  mainRes:
+    type: prov:index:Test
+	properties: %s`, propsJSON)
+
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", &schema.Provider{ResourcesMap: resMap})
+		pt := pulcheck.PulCheck(t, bridgedProvider, program)
+		_, err = pt.CurrentStack().Up(pt.Context())
+
+		require.Error(t, err)
```
Comment (t0yv0) [inline pkg/tests/schema_pulumi_test.go:4216]: Perhaps I missed this but does the type-checker error by default now? Or are we still in the mode where it was generating warnings. Just to make sure it's the type-checker catching this.

Comment (VenelinMartinov) [reply pkg/tests/schema_pulumi_test.go:4216]: Yeah, fair - it is a warning but something else fails in the provider. I'll change this to assert on the warning instead.

Comment (t0yv0) [reply pkg/tests/schema_pulumi_test.go:4216]: Out of scope for this PR but think about this if you are interested in this topic, I think we're very conservative with introducing new warnings but we don't assert on that in the tests. Possibly something in the test framework could be turned to strict mode by default to insist on no warnings or else on no new warnings except expected ones.

Comment (VenelinMartinov) [reply pkg/tests/schema_pulumi_test.go:4216]: I've seen TF tests have such an env var which turns all warnings into failures - we could add something like that for CI. It'd also give us additional testing on these type checker warnings as we'd see any failures in provider CI

**Pull request merged**